### PR TITLE
Agent: Make Hadoop interruptable

### PR DESCRIPTION
**Pull request by ilija-lazoroski**
# What does this PR do?

Makes Hadoop interuptable #1611.

Add any further explanations here.

## PR Checklist
* [x] Have you added an explanation of what your changes do and why you'd like to include them?
* [ ] ~Is the TravisCI build passing?~
* [ ] ~Was the CHANGELOG.md updated to reflect the changes?~
* [ ] ~Was the documentation framework updated to reflect the changes?~

## Testing Checklist

* [ ] ~Added relevant unit tests?~
* [x] Have you successfully tested your changes locally? Elaborate:
    > Tested by {Running the Monkey locally with relevant config/running Island/...}
* [x] If applicable, add screenshots or log transcripts of the feature working

## Explain Changes

Are the commit messages enough? If not, elaborate.

### CODE REVIEW

**Review by VakarisZ — CHANGES_REQUESTED** [review]
Looks good, but I don't understand why url building needs to be interruptable

File: monkey/infection_monkey/exploit/hadoop.py [hunk 1]
```diff
@@ -37,7 +37,7 @@ def __init__(self):
 
     def _exploit_host(self):
         # Try to get exploitable url
-        urls = self.build_potential_urls(self.host.ip_addr, self.HADOOP_PORTS)
+        urls = self.build_potential_urls(self.host.ip_addr, self.HADOOP_PORTS, self.interrupt)
```
Comment (mssalvatore) [inline monkey/infection_monkey/exploit/hadoop.py:40]: `build_potential_urls()` doesn't accept an interrupt.

File: monkey/infection_monkey/exploit/hadoop.py [hunk 2]
```diff
@@ -78,12 +82,21 @@ def exploit(self, url, command):
...
         resp = requests.post(
             posixpath.join(url, "ws/v1/cluster/apps/"), json=payload, timeout=LONG_REQUEST_TIMEOUT
         )
         return resp.status_code == 202
 
     def check_if_exploitable(self, url):
+        if self._is_interrupted():
+            self._set_interrupted()
+            return False
+
```
Comment (mssalvatore) [inline monkey/infection_monkey/exploit/hadoop.py:99]: `check_if_exploitable()` gets called from within a loop in `add_vulnerable_urls()`. That loop uses an interruptable iterator, so I think this check is redundant.

**Review by mssalvatore — APPROVED** [review]
Approved if tested

**Pull request merged**